uvm.go

```diff
@@ -0,0 +1,80 @@
+package hcsshim
+
+import (
+	"time"
+
+	"github.com/Microsoft/hcsshim/internal/schemaversion"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+// CreateUtilityVMOptions are the set of options which are passed to NewUtilityVM() to
+// create a utility vm.
+type CreateUtilityVMOptions struct {
+	Id                      string                  // Identifier for the uvm. Defaults to generated GUID.
+	Owner                   string                  // Specifies the owner. Defaults to executable name.
+	OperatingSystem         string                  // "windows" or "linux". // TODO: We can probably deduce this in a number of ways.
```

Comment:
We could detect Windows by the fact that LayerFolders has non-zero length. Not sure if it needs to be explicit.

Reply:
I think we should leave it explicit

---
In reply to: [189365952](https://github.com/Microsoft/hcsshim/pull/192#discussion_r189365952) [](ancestors = 189365952)